plugins/net_plugin/net_plugin.cpp

```diff
@@ -4333,47 +4443,105 @@ namespace eosio {
    }
 
    void connections_manager::connect_supplied_peers(const string& p2p_address) {
-      std::lock_guard g(connections_mtx);
       for (const auto& peer : supplied_peers) {
-         connect_i(peer, p2p_address);
+         resolve_and_connect(peer, p2p_address);
       }
    }
 
    void connections_manager::add( connection_ptr c ) {
       std::lock_guard g( connections_mtx );
-      add_i( std::move(c) );
+      boost::system::error_code ec;
+      auto endpoint = c->socket->remote_endpoint(ec);
+      connections.insert( connection_details{
+         .host = c->peer_address(), 
+         .c = std::move(c),
+         .active_ip = endpoint} );
    }
 
    // called by API
    string connections_manager::connect( const string& host, const string& p2p_address ) {
```

Comment:
```
   string net_plugin::connect( const string& host ) {
      return my->connections.connect( host, *my->p2p_addresses.begin() );
   }
```

`my->p2p_addresses` can be empty. For example:
```
./nodeos --data-dir d --config-dir d -e -p eosio --p2p-listen-endpoint "" --plugin eosio::net_api_plugin
curl http://127.0.0.1:8888/v1/net/connect -d '"127.0.0.1:9876"'
```
Will segfault.

Reply:
As above.